extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/common/GcpManager.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2023 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LCC - Initial implementation
+ *
+ */
+
+package org.eclipse.edc.gcp.common;
+
+import org.eclipse.edc.gcp.iam.IamService;
+import org.eclipse.edc.gcp.iam.IamServiceImpl;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+
+/**
+ * GCP manager object, provides connector configuration, IAM credentials.
+ */
+public class GcpManager {
+    private final ServiceExtensionContext context;
+    private final GcpConfiguration configuration;
+    private IamService iamService;
+
+    public GcpManager(ServiceExtensionContext context) {
+        this.context = context;
+        configuration = new GcpConfiguration(context);
+    }
+
+    public ServiceExtensionContext getContext() {
+        return context;
+    }
```

Comment:
definitely not needed, as the context is automatically injected to every extension.

Reply:
The idea is to inject the GcpManager object, and that extensions can use this single object to access the configuration . The assumption is that the context object doesn't change and is the same for every invocation of initialise method of the extension: is this the case?

Reply:
yes, exactly, is always the same, and generally you should get all the settings needed and compose the configuration object with those.
For reference, is described in the point 2 of the chapter `I` of the [coding principles document](https://github.com/eclipse-edc/docs/blob/main/developer/contributing/coding-principles.md#i-fail-fast-and-explicit-configuration):
> Settings can be pulled from the extension context and placed into configuration objects, which are passed to services via their constructor.

Reply:
Thanks for the reference, so the plan is to inject the GcpManager object to provide the GcpConfiguration, then the getContext() method can be skipped, will remove it.

Reply:
Done